aws_lambda_powertools/utilities/parameters/dynamodb.py

```diff
@@ -0,0 +1,164 @@
+        >>>
+        >>> ddb_provider.get("my-parameter")
+
+    **Retrieves a parameter value from a DynamoDB table in another AWS region**
+
+        >>> from botocore.config import Config
+        >>> from aws_lambda_powertools.utilities.parameters import DynamoDBProvider
+        >>>
+        >>> config = Config(region_name="us-west-1")
+        >>> ddb_provider = DynamoDBProvider("ParametersTable", config=config)
+        >>>
+        >>> ddb_provider.get("my-parameter")
+
+    **Retrieves a parameter value from a DynamoDB table passing options to the SDK call**
+
+        >>> from aws_lambda_powertools.utilities.parameters import DynamoDBProvider
+        >>> ddb_provider = DynamoDBProvider("ParametersTable")
+        >>>
+        >>> ddb_provider.get("my-parameter", ConsistentRead=True)
+
+    **Retrieves multiple values from a DynamoDB table**
+
+    In this case, the provider will use a sort key to retrieve multiple values using a query under
+    the hood. This expects that the sort key is named `sk`.
+
+        >>> from aws_lambda_powertools.utilities.parameters import DynamoDBProvider
+        >>> ddb_provider = DynamoDBProvider("ParametersTable")
+        >>>
+        >>> ddb_provider.get_multiple("my-parameters")
+
+    **Retrieves multiple values from a DynamoDB table with a custom sort key**
+
+    In this case, the provider will use a sort key to retrieve multiple values using a query under
+    the hood.
+
+        >>> from aws_lambda_powertools.utilities.parameters import DynamoDBProvider
+        >>> ddb_provider = DynamoDBProvider("ParametersTable")
+        >>>
+        >>> ddb_provider.get_multiple("my-parameters", sort_attr="my-sort-attr")
+
+    **Retrieves multiple values from a DynamoDB table passing options to the SDK calls**
+
+        >>> from aws_lambda_powertools.utilities.parameters import DynamoDBProvider
+        >>> ddb_provider = DynamoDBProvider("ParametersTable")
+        >>>
+        >>> ddb_provider.get("my-parameter", ConsistentRead=True)
+    """
+
+    table = None
+    key_attr = None
+    value_attr = None
+
+    def __init__(
+        self, table_name: str, key_attr: str = "id", value_attr: str = "value", config: Optional[Config] = None,
+    ):
+        """
+        Initialize the DynamoDB client
+        """
+
+        config = config or Config()
+        self.table = boto3.resource("dynamodb", config=config).Table(table_name)
+
+        self.key_attr = key_attr
+        self.value_attr = value_attr
+
+        super().__init__()
+
+    def _get(self, name: str, **sdk_options) -> str:
+        """
+        Retrieve a parameter value from Amazon DynamoDB
+
+        Parameters
+        ----------
+        name: str
+            Name of the parameter
+        sdk_options: dict, optional
+            Dictionary of options that will be passed to the get_item call
+        """
+
+        # Explicit arguments will take precedence over keyword arguments
+        sdk_options["Key"] = {self.key_attr: name}
```

Comment:
resolve me - Just wanted to drop saying I loved this new iteration ;)  It gives customers flexibility without taking the burden on maintaining that long term